lib/ModelMixins/CatalogMemberMixin.ts

```diff
@@ -11,6 +11,7 @@ type CatalogMember = Model<CatalogMemberTraits>;
 function CatalogMemberMixin<T extends Constructor<CatalogMember>>(Base: T) {
   abstract class CatalogMemberMixin extends AccessControlMixin(Base) {
     abstract get type(): string;
+    readonly typeName?: string;
```

Comment:
Every implementation of this is a getter, would be better to reflect that here?

Reply:
Yeah you're probably right, but wouldn't `readonly` be better than getters for properties which are set in the class definition and never change? 
Like `get type(): string` also

Reply:
Only reason I think typename could change is i18n but that might make more sense to map the type against translations rather than setting them on models